src/plans/stopping_criterion.jl

```diff
@@ -901,9 +901,20 @@ mutable struct StopWhenAny{TCriteria<:Tuple} <: StoppingCriterionSet
     StopWhenAny(c::Vector{StoppingCriterion}) = new{typeof(tuple(c...))}(tuple(c...), "")
     StopWhenAny(c::StoppingCriterion...) = new{typeof(c)}(c, "")
 end
+
+@inline _fast_any(f, tup::Tuple{}) = true
+@inline _fast_any(f, tup::Tuple{T}) where {T} = f(tup[1])
+@inline function _fast_any(f, tup::Tuple)
+    if f(tup[1])
+        return true
+    else
+        return _fast_any(f, tup[2:end])
+    end
+end
+
 function (c::StopWhenAny)(p::AbstractManoptProblem, s::AbstractManoptSolverState, i::Int)
     (i == 0) && (c.reason = "") # reset on init
-    if any(subC -> subC(p, s, i), c.criteria)
+    if _fast_any(subC -> subC(p, s, i), c.criteria)
```

Comment:
The standard Julia implementation of `any` was costly enough that it was one of the biggest things on the profiling plot. Even my very simple replacement is faster :shrug: